slack-code-runner-bot/function.js

```diff
@@ -0,0 +1,123 @@
+const { WebClient } = require('@slack/client');
+const { VM } = require('vm2');
+const decode = require('decode-html');
+
+const {
+  SLACK_BOT_TOKEN: token,
+  BOT_TRIGGER_STATEMENT: triggerStatement,
+  MAX_CODE_RUNTIME_MS: maxTimeMS,
+} = process.env;
+const web = new WebClient(token);
+
+const triBacktick = '```';
+
+/**
+ * Extracts the potential JavaScript code segment located
+ * between leading and trailing triple backticks ```
+ *
+ * @param {string} textToExtractFrom - segment to attempt code extraction on
+ *
+ * @return {string} - extracted code segment
+ */
+function extractBetweenBackticks(textToExtractFrom) {
+  const startIdx = textToExtractFrom.indexOf(triBacktick);
+  const endIdx = textToExtractFrom.lastIndexOf(triBacktick);
+
+  if (startIdx === -1 || endIdx === -1 || (startIdx === endIdx)) {
+    throw new Error('Javascript input must begin and end with triple backticks');
+  }
+  return extractText.substring(startIdx + 3, endIdx);
+}
```

Comment:
Would this function be easier to write with a regexp ``/\`\`\`(.*)\`\`\`/``?  (*Please* say "yes", marking this regexp down was incredibly tricky...).

Reply:
It's funny, I originally had it as regexp and Michael suggested I change it in slack. I should have posted those comments he made in this PR (I've done that now).